R/ael02.R

```diff
@@ -0,0 +1,69 @@
+# ael02_main ----
+
+#' @describeIn ael02 Main TLG function
+#'
+#' @inheritParams gen_args
+#' @returns the main function returns an `rlistings` or a `list` object.
+#' @export
+#'
+ael02_main <- modify_default_args(std_listing,
+  dataset = "adae",
+  key_cols = c("ID", "ASR", "ACTARM"),
+  disp_cols = c(
+    "AEDECOD", "TRTSDTM", "ASTDY", "ADURN", "AESER",
+    "ASEV", "AREL", "AEOUT", "AECONTRT", "AEACN"
+  )
+)
+
+#' @describeIn ael02 Preprocessing
+#'
+#' @inheritParams ael02_main
+#' @inheritParams gen_args
+#' @returns the preprocessing function returns a `list` of `data.frame`.
+#' @export
+#'
+ael02_pre <- function(adam_db,
+                      dataset = "adae",
+                      arm_var = "ACTARM",
+                      ...) {
+  adam_db[[dataset]] <- adam_db[[dataset]] %>%
+    filter(.data$ANL01FL == "Y") %>%
+    mutate(
+      across(
+        all_of(c(arm_var, "AEDECOD", "ASEV", "AEOUT", "AEACN")),
+        ~ reformat(.x, missing_rule)
+      )
+    ) %>%
+    mutate(
+      !!arm_var := with_label(.data[[arm_var]], "Treatment"),
+      ID = create_id_listings(.data$SITEID, .data$SUBJID),
```

Comment:
why use SUBJID instead of USUBJID?

Reply:
It was in the [specs ](https://github.com/insightsengineering/chevron/issues/643) but I can change that.